autoPyTorch/api/base_task.py

```diff
@@ -1406,10 +1350,27 @@ def refit(
             self
         """
 
+        if dataset is None:
+            if (
+                X_train is None
+                and y_train is None
+            ):
+                raise ValueError("No dataset provided, must provide X_train, y_train tensors")
```

Comment:
Can we move those lines to `self.get_dataset`?

Reply:
I think `self.get_dataset` is exposed to the users and they should be able to use it without confusion. To move these lines, we would have to allow X_train and y_train to be None for mypy and then raise this error. I think it's fine this way.